lib/logger/logger.client.ts

```diff
@@ -0,0 +1,72 @@
+import { createLoggerHelpers, createNamedLoggers } from './utils/core';
+
+/**
+ * Type definition for browser-safe logger
+ * Implements a subset of pino.Logger interface using console methods
+ */
+export interface BrowserLogger {
+  trace: (...args: unknown[]) => void;
+  debug: (...args: unknown[]) => void;
+  info: (...args: unknown[]) => void;
+  warn: (...args: unknown[]) => void;
+  error: (...args: unknown[]) => void;
+  fatal: (...args: unknown[]) => void;
+  child: (bindings: Record<string, unknown>) => BrowserLogger;
+}
+
+/**
+ * Browser-safe logger that prevents errors when running client-side
+ * Uses console methods as fallback in browser environment
+ * @param accumulatedBindings - Bindings accumulated from parent loggers
+ */
+export function createBrowserLogger(
+  accumulatedBindings: Record<string, unknown> = {}
+): BrowserLogger {
+  // Create context string from accumulated bindings (empty string if no bindings)
+  const contextStr =
+    Object.keys(accumulatedBindings).length > 0 ? JSON.stringify(accumulatedBindings) : '';
+
+  return {
+    trace: (...args: unknown[]) =>
+      contextStr ? console.trace(contextStr, ...args) : console.trace(...args),
+    debug: (...args: unknown[]) =>
+      contextStr ? console.debug(contextStr, ...args) : console.debug(...args),
+    info: (...args: unknown[]) =>
+      contextStr ? console.info(contextStr, ...args) : console.info(...args),
+    warn: (...args: unknown[]) =>
+      contextStr ? console.warn(contextStr, ...args) : console.warn(...args),
+    error: (...args: unknown[]) =>
+      contextStr ? console.error(contextStr, ...args) : console.error(...args),
```

Comment:
It's useful to see the type of console.log at a glance, because most browsers don't display the different types in different ways

```suggestion
    trace: (...args: unknown[]) =>
      contextStr ? console.trace(contextStr, ...args) : console.trace(...args),
    debug: (...args: unknown[]) =>
      contextStr ? console.debug('[DEBUG]', contextStr, ...args) : console.debug('[DEBUG]', ...args),
    info: (...args: unknown[]) =>
      contextStr ? console.info(contextStr, ...args) : console.info(...args),
    warn: (...args: unknown[]) =>
      contextStr ? console.warn(contextStr, ...args) : console.warn(...args),
    error: (...args: unknown[]) =>
      contextStr ? console.error(contextStr, ...args) : console.error(...args),

Reply:
@4fra I know I mentioned this before, but I think we both misunderstood each other, because I believe you were thinking more server-side logs, and I was thinking more browser logs

Reply:
@4fra now I made it only for [DEBUG], because that's the only one which looks the same as INFO in chrome console

Reply:
## Adding prefixes defeats the purpose of native functions.

Yes, console.debug() and console.info() look identical in Chromium Console, this is intentional browser design. The distinction is achieved through filtering, not visual styling.

**Instead of prefixes, developers should use filtering.**

### Cons of the proposed change
- Visual Noise: Prefixes add clutter to logs, making them harder to scan quickly
- Native Filtering: Browsers provide built-in log level filtering (Verbose, Info, Warnings, Errors) which is more powerful than visual prefixes
- Console Grouping: Chrome DevTools allows filtering by regex, so you can search for specific log patterns without needing prefixes
- Performance: Extra strings in every log call add unnecessary overhead
- Industry Standard: Popular browser logging libraries (like debug, loglevel, winston-browser) don't use prefixes in their default configurations

<img width="1280" height="720" alt="image" src="https://github.com/user-attachments/assets/4bdd64b6-1780-45d5-9354-83c318975675" />

https://www.youtube.com/watch?v=yCLjkXrChVE

### Major open-source projects like next.js, react-devtools or vite use `console.debug()` **without** prefixes.